src/components/input/input.js

```diff
@@ -777,14 +777,6 @@ function ngMessagesAnimation($q, $animateCss) {
 function ngMessageAnimation($animateCss) {
   return {
     enter: function(element, done) {
-      var messages = getMessagesElement(element);
-
-      // If we have the md-auto-hide class, the md-input-invalid animation will fire, so we can skip
-      if (messages.hasClass('md-auto-hide')) {
-        done();
-        return;
```

Comment:
Yes, this lines are required, otherwise the input container will fire the enter animations twice. 
 > https://github.com/angular/material/pull/7583/files